src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,45 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private CopyOnWriteArrayList<EventListener> events = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor;
 
+    public EventManager() {
+        this.events = new CopyOnWriteArrayList<>();
+        this.executor = Executors.newCachedThreadPool();
     }
 
-    public void deregisterListener(EventListener listener) {
+    public void registerListener(EventListener listener) {
+        events.add(listener);
+    }
 
+    public void deregisterListener(EventListener listener) {
+        events.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : events) {
```

Comment:
Using an ExecutorService to submit tasks that execute 'onEvent' is a good practice for asynchronous processing. However, be aware that using 'newCachedThreadPool' can potentially create a large number of threads if many events are being notified rapidly, which might lead to performance issues. If you expect a high volume of events, consider using a fixed thread pool or another type of executor that limits the number of concurrent threads.